examples/farmer.js

```diff
@@ -0,0 +1,72 @@
+const Vec3 = require('vec3')
+const mineflayer = require('mineflayer')
+
+if (process.argv.length < 4 || process.argv.length > 6) {
+  console.log('Usage : node scoreboard.js <host> <port> [<name>] [<password>]')
+  process.exit(1)
+}
+
+const bot = mineflayer.createBot({
+  host: process.argv[2],
+  port: parseInt(process.argv[3]),
+  username: process.argv[4] ? process.argv[4] : 'farmer',
+  password: process.argv[5],
+  verbose: true
+})
+
+// To fish we have to give bot the seeds
+// /give farmer wheat_seeds 64
+
+const ID_FARMLAND = 60
+const ID_WHEAT = 59
+const ID_SEEDS = 295
+
+function blockToSow () {
+  return bot.findBlock({
+    point: bot.entity.position,
+    matching: (block) => {
+      if (block && block.type === ID_FARMLAND) {
+        const blockAbove = bot.blockAt(block.position.offset(0, 1, 0))
+        return !blockAbove || blockAbove.type === 0
+      }
+
+      return false
+    }
+  })
+}
+
+function blockToHarvest () {
+  return bot.findBlock({
+    point: bot.entity.position,
+    matching: (block) => {
+      return block && block.type === ID_WHEAT && block.metadata === 7
+    }
+  })
+}
+
+function loop () {
+  try {
+    const toHarvest = blockToHarvest()
+    if (toHarvest) {
+      return bot.dig(toHarvest, () => {
+        return setTimeout(loop, 1000)
```

Comment:
Why are these setTimeout needed ?

Reply:
Without the timeouts we could get `Maximum call stack size exceeded` error

Reply:
Ok, do setTimeout(loop, 0)

Putting arbitrary timeout values (non 0) tend to make things work "most of the time" which is not great for the times when it doesn't work.

Reply:
setImmediate(loop) is just the same as setTimeout(loop, 0)